lib/checks/shared/filename-is-valid-accessible-name.js

```diff
@@ -0,0 +1,29 @@
+const { text } = axe.commons;
+const possibleImageExtensions = [`.jpg`, `.jpeg`, `.png`, `.gif`, `.tif`];
+
+const accText = text.accessibleText(node).toLowerCase();
+
+/**
+ * Check if accessible name, uses filename (along with file extension)
+ * eg: Download image of Tajmahal.jpg
+ */
+const fileExtnInAccText = accText.match(/\.[0-9a-z]+$/i);
+if (
+	fileExtnInAccText &&
+	possibleImageExtensions.includes(fileExtnInAccText[0])
+) {
+	return undefined;
+}
+
+/**
+ * Get source value & check of filename is used in accessible name
+ */
+const srcValue = node.getAttribute('src').toLowerCase();
+const fileName = text.sanitize(srcValue.split('/').pop());
+const fileNameWithoutExtn = fileName.split('.').shift();
+
+if (accText.includes(fileNameWithoutExtn)) {
+	return true;
+}
+
+return undefined;
```

Comment:
This will return a `needs review` for every alt attribute that is not a file name without the extension. As far as I can tell, this would return a needs review for

`<img alt="Favorite" src="star.jpg" />`

This is absolutely terrible!